src/ck-core/ck.C

```diff
@@ -3055,13 +3077,13 @@ void CkMessageWatcherInit(char **argv,CkCoreState *ck) {
 	}
 }
 
-int CkMessageToEpIdx(void *msg) {
-        envelope *env=UsrToEnv(msg);
-	int ep=env->getEpIdx();
-	if (ep==CkIndex_CkArray::recvBroadcast(0))
-		return env->getsetArrayBcastEp();
-	else
-		return ep;
+int CkMessageToEpIdx(void* msg) {
+  envelope* env = UsrToEnv(msg);
+  int type = env->getMsgtype();
```

Comment:
Probably won't matter in this case, but shouldn't these convenience stack temporaries (i.e., type) be const or even const reference?

Reply:
Couldn't hurt, but it could be a wider-ranging cleanup effort and it probably isn't worth holding back the PR.